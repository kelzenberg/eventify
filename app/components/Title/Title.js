import React from 'react';
import PropTypes from 'prop-types';

export default function Header(props) {
    let breadcrumbs = props.breadcrumbs == undefined ? [] : props.breadcrumbs;
    return <div className="container mb-4 mt-5">
        <span className="text-muted">{breadcrumbs.join(" > ")}</span>
        <div className="d-flex">
            <h1 className="fw-bolder me-auto">{props.title}</h1>
            <div className="">
                {props.children}
            </div>
        </div>
    </div>
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    breadcrumbs: PropTypes.arrayOf(PropTypes.string)
}