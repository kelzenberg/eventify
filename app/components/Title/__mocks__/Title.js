import React from "react";

export default jest.fn((props) => {return <div>Mocked Title Component: {props.children}</div>});