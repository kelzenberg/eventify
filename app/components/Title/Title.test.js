import React from 'react';
import {render} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Title from "./Title";
import { MemoryRouter } from 'react-router-dom';

test('Title', () => {
    // MemoryRouter is needed for the <Link/> components
    let { container, rerender } = render(
        <MemoryRouter>
            <Title title="Test Title" breadcrumbs={["one", {name: "two", link: "/linkToTwo"}, undefined]}>
                <span>child content</span>
            </Title>
        </MemoryRouter>
    );
    // This component is so simple a single snapshot is enough
    expect(container).toMatchSnapshot();

    // without breadcrumbs
    rerender(
        <MemoryRouter>
            <Title title="Test Title">
                <span>child content</span>
            </Title>
        </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
})