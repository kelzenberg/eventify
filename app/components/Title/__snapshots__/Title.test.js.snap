// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title 1`] = `
<div>
  <div
    class="container-lg mb-4 mt-5 title"
  >
    <span
      class="text-muted"
    >
      <span>
        one
      </span>
       &gt; 
      <a
        href="/linkToTwo"
      >
        two
      </a>
       &gt; 
      <span />
    </span>
    <div
      class="d-flex"
    >
      <h1
        class="fw-bolder me-auto"
      >
        Test Title
      </h1>
      <div
        class=""
      >
        <span>
          child content
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Title 2`] = `
<div>
  <div
    class="container-lg mb-4 mt-5 title"
  >
    <span
      class="text-muted"
    />
    <div
      class="d-flex"
    >
      <h1
        class="fw-bolder me-auto"
      >
        Test Title
      </h1>
      <div
        class=""
      >
        <span>
          child content
        </span>
      </div>
    </div>
  </div>
</div>
`;
